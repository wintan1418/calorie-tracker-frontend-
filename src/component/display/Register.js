import { Link, useHistory } from 'react-router-dom';

const Register = () => {
  const history = useHistory();

  async function postData(url = '', data = {}) {
    const response = await fetch(url, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(data),
    });
    return response.json()
      .then((data) => {
        console.log(data)
        sessionStorage.setItem('token', JSON.stringify(data.auth_token));
        history.push('/assesment');
      });
  }
  const handleRegister = async (e) => {
    e.preventDefault();
    const userInput = {
      username: e.target[0].value,
      email: e.target[1].value,
      password: e.target[2].value,
      present_measure: e.target[3].value,
      proposed_measure: e.target[4].value,
      age: e.target[5].value,
      sex: e.target[6].value
  
      
    };

    const urlregister = 'https://juwons.herokuapp.com/register';
    postData(urlregister, userInput);
  };
  return (
    <div className="authen-page">
      <div>
        <i className="fas fa-user-circle" />
      </div>
      <form onSubmit={(e) => handleRegister(e)}>
        <input type="text" placeholder="Your username please" required />
        <input type="email" placeholder="Your email please" required />
        <input type="password" placeholder="Password" required />
        <input type="age" placeholder="Your age please" required />
        <input type="sex" placeholder="male/female" required />
      

        <label htmlFor="present_measure">
          <span>Your current calorie  measure</span>
          <input type="number" step=".5" min="0" max="20" name="present_measure" id="present_measure" placeholder="Enter 0 ~ 9" required />
        </label>

        <label htmlFor="proposed_measure">
          <span>Your proposed calorie measure</span>
          <input type="number" step=".5" min="0" max="20" name="proposed_measure" id="proposed_measure" placeholder="Enter 0 ~ 9" required />
        </label>

        <button type="submit">Submit</button>
      </form>
      <div className="authen-text">
        Back to
        <Link to="/log_in">Log in page</Link>
      </div>
    </div>
  );
};

export default Register;