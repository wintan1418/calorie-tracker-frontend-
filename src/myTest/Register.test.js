import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Register from  '../component/display/Register';

it('renders accurately', () => {
  const tree = renderer.create(
    <BrowserRouter>
    <Register />
    </BrowserRouter>,
  ).toJSON();
  expect(tree).toMatchSnapshot();

});