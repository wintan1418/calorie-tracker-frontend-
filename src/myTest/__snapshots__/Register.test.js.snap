// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders accurately 1`] = `
<div
  className="authen-page"
>
  <div>
    <i
      className="fas fa-user-circle"
    />
  </div>
  <form
    onSubmit={[Function]}
  >
    <input
      placeholder="Your username please"
      required={true}
      type="text"
    />
    <input
      placeholder="Your email please"
      required={true}
      type="email"
    />
    <input
      placeholder="Password"
      required={true}
      type="password"
    />
    <input
      placeholder="Your age please"
      required={true}
      type="age"
    />
    <input
      placeholder="male/female"
      required={true}
      type="sex"
    />
    <label
      htmlFor="present_measure"
    >
      <span>
        Your current calorie  measure
      </span>
      <input
        id="present_measure"
        max="20"
        min="0"
        name="present_measure"
        placeholder="Enter 0 ~ 9"
        required={true}
        step=".5"
        type="number"
      />
    </label>
    <label
      htmlFor="proposed_measure"
    >
      <span>
        Your proposed calorie measure
      </span>
      <input
        id="proposed_measure"
        max="20"
        min="0"
        name="proposed_measure"
        placeholder="Enter 0 ~ 9"
        required={true}
        step=".5"
        type="number"
      />
    </label>
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
  <div
    className="authen-text"
  >
    Back to
    <a
      href="/log_in"
      onClick={[Function]}
    >
      Log in page
    </a>
  </div>
</div>
`;
